flight/targets/sprf3e/fw/pios_board.c

```diff
@@ -0,0 +1,479 @@
+                         NULL);                   // sbus_cfg
+
+	uint8_t hw_uart2;
+	HwSprf3eUart2Get(&hw_uart2);
+	PIOS_HAL_ConfigurePort(hw_uart2,              // port_type
+                         &pios_uart2_cfg,         // usart_port_cfg
+                         &pios_usart_com_driver,  // com_driver
+                         NULL,                    // i2c_id
+                         NULL,                    // i2c_cfg
+                         &pios_ppm_cfg,           // ppm_cfg
+                         NULL,                    // pwm_cfg
+                         PIOS_LED_ALARM,          // led_id
+                         &pios_uart2_dsm_aux_cfg, // dsm_cfg
+                         hw_DSMxMode,             // dsm_mode
+                         NULL);                   // sbus_cfg
+
+	uint8_t hw_uart1;
+	HwSprf3eUart1Get(&hw_uart1);
+	PIOS_HAL_ConfigurePort(hw_uart1,             // port_type
+                         &pios_uart1_cfg,  // usart_port_cfg
+                         &pios_usart_com_driver,  // com_driver
+                         NULL,                    // i2c_id
+                         NULL,                    // i2c_cfg
+                         NULL,                    // ppm_cfg
+                         NULL,                    // pwm_cfg
+                         PIOS_LED_ALARM,          // led_id
+                         &pios_uart1_dsm_aux_cfg, // dsm_cfg
+                         hw_DSMxMode,             // dsm_mode
+                         NULL);                   // sbus_cfg
+
+#if defined(PIOS_INCLUDE_GCSRCVR)
+	GCSReceiverInitialize();
+	uintptr_t pios_gcsrcvr_id;
+	PIOS_GCSRCVR_Init(&pios_gcsrcvr_id);
+	uintptr_t pios_gcsrcvr_rcvr_id;
+	if (PIOS_RCVR_Init(&pios_gcsrcvr_rcvr_id, &pios_gcsrcvr_rcvr_driver, pios_gcsrcvr_id)) {
+		PIOS_Assert(0);
+	}
+	pios_rcvr_group_map[MANUALCONTROLSETTINGS_CHANNELGROUPS_GCS] = pios_gcsrcvr_rcvr_id;
+#endif	/* PIOS_INCLUDE_GCSRCVR */
+
+#ifndef PIOS_DEBUG_ENABLE_DEBUG_PINS
+#ifdef PIOS_INCLUDE_SERVO
+	PIOS_Servo_Init(&pios_servo_cfg);
+#endif
+#else
+	PIOS_DEBUG_Init(&pios_tim_servo_all_channels, NELEMENTS(pios_tim_servo_all_channels));
+#endif
+
+#if defined(PIOS_INCLUDE_ADC)
+	uint32_t internal_adc_id;
+	if (PIOS_INTERNAL_ADC_Init(&internal_adc_id, &internal_adc_cfg) < 0)
+		PIOS_Assert(0);
+	PIOS_ADC_Init(&pios_internal_adc_id, &pios_internal_adc_driver, internal_adc_id);
+#endif /* PIOS_INCLUDE_ADC */
+	PIOS_WDG_Clear();
+	PIOS_DELAY_WaitmS(200);
+	PIOS_WDG_Clear();
+
+#if defined(PIOS_INCLUDE_BMP280)
+#include "pios_bmp280.h"
+#include "pios_bmp280_priv.h"
+  uint8_t hw_baro;
+  HwSprf3eBarometerGet(&hw_baro);
+    switch(hw_baro) {
+        case HWSPRF3E_BAROMETER_BMP280:
+        	if (PIOS_HAL_ConfigureExternalBaro(hw_baro,
+			       &pios_i2c_internal_id, &pios_i2c_internal_cfg))
+      			PIOS_SENSORS_SetMissing(PIOS_SENSOR_BARO);
+
+/*        	if ((PIOS_BMP280_Init(&pios_bmp280_cfg, pios_i2c_internal_id) != 0)
```

Comment:
Should the memory not be freed, if init fails? At the moment baro can be selected between BMP280 and none. If "none" is configured, I will not init the baro. If BMP280 is selected, I try to init the baro. So there should be no waste of memory? Since where is only one I2C, it is the same if the baro is onboard or external connected.